packages/remirror__extension-link/src/link-extension.ts

```diff
@@ -739,6 +835,91 @@ export class LinkExtension extends MarkExtension<LinkOptions> {
 
     return autoLinkAllowedTLDs.includes(tld);
   }
+
+  private findURL(input: string): string | undefined {
+    for (const match of findMatches(input, this.options.autoLinkRegex)) {
+      const urlMatch = getMatchString(match);
+
+      if (!this.isValidUrl(urlMatch)) {
+        continue;
+      }
+
+      return urlMatch;
+    }
+
+    return;
+  }
+
+  private getURLEndIndex({
+    url,
+    startIndex = 0,
+    text,
+  }: {
+    url?: string;
+    startIndex?: number;
+    text: string;
+  }) {
+    const newUrl = createNewURL(text);
+
+    // Check for trailing punctuation and adjacent characters
+    if (newUrl) {
```

Comment:
Instead of wrapping the entire block, I would prefer exiting early

```suggestion
    if (!newUrl) {
      return;
    }
```

Reply:
Yes I should have done that